pkg/runtime/bulksub_resiliency.go

```diff
@@ -0,0 +1,63 @@
+package runtime
+
+import (
+	"context"
+	"sync/atomic"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/pkg/errors"
+
+	"github.com/dapr/components-contrib/pubsub"
+	"github.com/dapr/dapr/pkg/resiliency"
+	"github.com/dapr/dapr/utils"
+)
+
+// ApplyBulkSubscribeResiliency applies resiliency support to bulk subscribe. It tries to filter
+// out the messages that have been successfully processed and only retries the ones that have failed
+func (a *DaprRuntime) ApplyBulkSubscribeResiliency(bulkSubCallData *bulkSubscribeCallData,
+	psm pubsubBulkSubscribedMessage, deadLetterTopic string, path string, policyDef *resiliency.PolicyDefinition,
+	rawPayload bool, envelope map[string]interface{},
+) (*[]pubsub.BulkSubscribeResponseEntry, error) {
+	bscData := *bulkSubCallData
+	ctx := bscData.ctx
+	var subscribedEntries atomic.Pointer[[]pubSubMessage]
+	subscribedEntries.Store(&psm.pubSubMessages)
```

Comment:
Why is this using atomic? I see it's only used in the accumulator, so it shouldn't need to be atomic anymore, unless I misunderstand

Reply:
Yeah, correct - atomic shouldn't be required - changing that.